app/src/main/java/com/karumi/data/repository/NetworkSuperHeroDataSource.kt

```diff
@@ -0,0 +1,65 @@
+package com.karumi.data.repository
+
+import com.karumi.domain.model.AuthDomainError
+import com.karumi.domain.model.DomainError
+import com.karumi.domain.model.NotInternetDomainError
+import com.karumi.domain.model.SuperHero
+import com.karumi.domain.model.UnknownDomainError
+import com.karumi.marvelapiclient.CharacterApiClient
+import com.karumi.marvelapiclient.MarvelAuthApiException
+import com.karumi.marvelapiclient.model.CharacterDto
+import com.karumi.marvelapiclient.model.MarvelImage
+import com.karumi.marvelapiclient.model.MarvelResponse
+import org.funktionale.either.Either
+import java.net.ConnectException
+
+class NetworkSuperHeroDataSource(val apiClient: CharacterApiClient) : SuperHeroDataSource {
+    companion object {
+        val PAGE_SIZE = 10
+        val RESPONSE_OK = 200
+    }
+
+    override fun get(key: String): Either<DomainError, SuperHero> =
+        try {
+            mapResponse(apiClient.getCharacter(key), { mapSuperHero(it) })
```

Comment:
There is a feature in Kotlin for this sort of functions (where the last parameter is a function) that allows you to put the function out of the parens:

`mapResponse(apiClient.getCharacter(key), { mapSuperHero(it) })` => `mapResponse(apiClient.getCharacter(key)) { mapSuperHero(it) }`

Besides that, I'd rather separate in a different line the API call so that it's more readable (looking at the function it seems like it's mapping something but the main action we are doing is calling the `apiClient`) but that's a matter of taste.

Reply:
call I forgot this property, thank you